x-pack/plugins/ingest_manager/server/services/index_pattern.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { SavedObjectsClientContract } from 'kibana/server';
+import * as Registry from './epm/registry';
+import { getInstallationObject, findInstalledPackageByName } from './epm/packages/get';
+import { Installation } from '../types';
+
+export interface IndexPatternService {
+  get(
+    savedObjectsClient: SavedObjectsClientContract,
+    pkgName: string,
+    datasetPath: string,
+    version?: string
+  ): Promise<string | undefined>;
+}
+
+export class IndexPatternSavedObjectService implements IndexPatternService {
+  public async get(
+    savedObjectsClient: SavedObjectsClientContract,
+    pkgName: string,
+    datasetPath: string,
+    version?: string
+  ): Promise<string | undefined> {
+    let installation: Installation | undefined;
+    if (version) {
+      const pkgkey = Registry.pkgToPkgKey({
```

Comment:
I had been wondering if a utility like this (`pkgToPkgKey()` existed - now I know it does.